src/Console/Command/AnalyseCommand.php

```diff
@@ -76,34 +75,28 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         ini_set('memory_limit', '-1');
 
         $symfonyOutput = new SymfonyOutput($output, new Style(new SymfonyStyle($input, $output)));
-
-        $file = $input->getArgument('depfile') ?? $this->getDefaultFile($symfonyOutput);
-
-        if (!is_string($file)) {
-            throw InvalidArgumentException::invalidDepfileType($file);
-        }
-
-        $configuration = $this->configurationLoader->load($file);
+        /** @var string[] $formatters */
+        $formatters = (array) $input->getOption('formatter');
+        $options = new AnalyseOptions(
+            $input->getArgument('depfile') ?? $this->getDefaultFile($symfonyOutput),
+            (bool) $input->getOption('no-progress'),
+            $formatters,
+            (bool) $input->getOption(self::OPTION_FAIL_ON_UNCOVERED)
+        );
 
         $this->dispatcher->addSubscriber(new ConsoleSubscriber($output));
-
-        if (!$input->getOption('no-progress')) {
+        if ($options->showProgress()) {
             $this->dispatcher->addSubscriber(new ProgressSubscriber($symfonyOutput));
         }
 
+        $configuration = $this->configurationLoader->load($options->getConfigurationFile());
+
         $this->printCollectViolations($symfonyOutput);
         $context = $this->analyser->analyse($configuration);
 
         $this->printFormattingStart($symfonyOutput);
 
-        /** @var string[] $formats */
-        $formats = (array) $input->getOption('formatter');
-
-        $formatters = [];
-        if ($formats) {
```

Comment:
This condition was removed, because `getFormattersByNames` can work with `[]` as input.